### PR TITLE
Add the possibility to copy EXIF tags

### PR DESCRIPTION
**Pull request by patrickfournier**
If IMAGE_PROCESS_COPY_EXIF_TAGS is True, use exiftool to copy tags from source image to transformed images.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -472,6 +472,15 @@ opens your files. The default is `utf-8`.
 IMAGE_PROCESS_ENCODING = "utf-8"
 ```
 
+#### Copying EXIF tags
+
+You may ask `image_process` to copy the EXIF tags from your original image to
+the transformed images. You must have [exiftool](https://exiftool.org/) installed.
```
Comment (justinmayer) [inline README.md:478]: For consistency, I suggest changing: \`image_process\` ⇢ `_Image Process_`

File: README.md [hunk 2]
```diff
@@ -472,6 +472,15 @@ opens your files. The default is `utf-8`.
 IMAGE_PROCESS_ENCODING = "utf-8"
 ```
 
+#### Copying EXIF tags
```
Comment (justinmayer) [inline README.md:475]: tags ⇢ Tags

File: pelican/plugins/image_process/image_process.py [hunk 3]
```diff
@@ -220,6 +296,10 @@ def harvest_images_in_fragment(fragment, settings):
     parser = settings.get("IMAGE_PROCESS_PARSER", "html.parser")
     soup = BeautifulSoup(fragment, parser)
 
+    copy_exif_tags = settings.get("IMAGE_PROCESS_COPY_EXIF_TAGS", False)
+    if copy_exif_tags:
+        ExifTool.instance = ExifTool()
```
Comment (justinmayer) [inline pelican/plugins/image_process/image_process.py:301]: Perhaps somewhere around here we should also check to make sure the `exiftool` executable is actually installed? Something like:
```python
import shutil

if shutil.which("exiftool") is None:
    # [print user-friendly warning that EXIF tags were not copied due to missing `exiftools`]
else:
    […]
```

**Review by justinmayer — COMMENTED** [review]
Many thanks for adding EXIF tag capability to this plugin. I made a few minor suggestions. Would you have a look and let me know what you think?

File: README.md (hunk 2 above)
Comment (patrickfournier) [reply README.md]: fixed

File: README.md (hunk 1 above)
Comment (patrickfournier) [reply README.md:478]: fixed

File: pelican/plugins/image_process/image_process.py (hunk 3 above)
Comment (patrickfournier) [reply pelican/plugins/image_process/image_process.py]: fixed

**Comment by justinmayer** [comment]
Great. Thanks, Patrick. Would it be too much to ask for one or more tests for this new feature? That way we can have more assurance that your helpful feature is not broken by some future change to the plugin's behavior. 😊

**Comment by patrickfournier** [comment]
@justinmayer Sure, I'll add a test. Thanks for your review!

**Comment by patrickfournier** [comment]
@justinmayer I added two tests. I am not very familiar with the test framework so do not hesitate to point out any misuse or abuse.

**Comment by kernc** [comment]
Maybe install exiftool [on CI](https://github.com/pelican-plugins/image-process/blob/main/.github/workflows/main.yml) so the test runs instead of skipping?

**Comment by patrickfournier** [comment]
@kernc good point! Done

**Review by justinmayer — APPROVED** [review]
Looks good!

**Comment by justinmayer** [comment]
Many thanks to @patrickfournier for the nice enhancement and to @kernc for helping to review. 💫

**Comment by justinmayer** [comment]
✨ This feature was just released in Image Process 2.1  ✨

**Comment by patrickfournier** [comment]
Thanks for the release @justinmayer !

**Pull request merged**